crucible-mir/CHANGELOG.md

```diff
@@ -1,5 +1,8 @@
 # 0.3 -- 2024-08-30
 
+* The calling sequence of ```translateMIR``` has changed: the first argument,
+  which should always have been passed as ```mempty```, has been removed.
+  This will probably require adjustments in downstream callers.
```

Comment:
We already assigned a date (2024-08-30) to the 0.3 release, so let's put this bullet point under a new section (with the head `# next -- TBA`) above this one.

Reply:
Also, make sure to advertise the removal of the `Semigroup` and `Monoid` instances, since those are also part of the public API.

Reply:
I've moved the entry in the right place and added the instances (but there are not expected to be any downstream users -- are these part of the public API because you can't hide instances, or were they intended? (And is there any better way to determine what's in the public API than wade through the export lists of every source file? Is the ```Pass``` type in Mir.Pass, which has one fewer implicit argument now, also part of the public API because it's apparently exported and all the source files are exposed?)

Reply:
> are these part of the public API because you can't hide instances, or were they intended?

Instances are always exported, so they're always part of the public API. More specifically...

> And is there any better way to determine what's in the public API than wade through the export lists of every source file? 

My measuring stick for what constitutes a publicly-facing API change is to use the [Haskell Package Versioning Policy (PVP)](https://pvp.haskell.org/). According to the PVP, removing an instance is always a breaking change.

> Is the `Pass` type in Mir.Pass, which has one fewer implicit argument now, also part of the public API because it's apparently exported and all the source files are exposed?

While the PVP doesn't say anything about type synonyms per se, since we are changing the definition of `Pass`, I think one could argue that that is a breaking change per the PVP. We could mention this in the changelog to be on the safe side.